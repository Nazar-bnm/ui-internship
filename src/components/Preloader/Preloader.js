import React from 'react';
import PropTypes from 'prop-types';

import './Preloader.scss';

import circle from '../../assets/img/preloader/circle.svg';

const Preloader = (props) => {
  const { overlay, size } = props;
  return (
    <div className={overlay ? 'preloader-wrapper overlayStyle' : 'preloader-wrapper'}>
      <div className="preloader">
        <img src={circle} alt="preloder" style={{ height: size }} />
      </div>
    </div>
  );
};

Preloader.propTypes = {
  overlay: PropTypes.string
};

Preloader.defaultProps = {
  overlay: null
};

export default Preloader;
