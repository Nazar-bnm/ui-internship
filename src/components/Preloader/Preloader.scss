@import '../../scss/global_styles';

.preloader {
  top: 50%;
  position: absolute;
  z-index: 2;
  left: 50%;
  transform: translate(-50%, -50%);

  &-wrapper {
    background-color: rgba(255,255,255,0.8);
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 1;
  }

}
    .overlayStyle {
      position: fixed;
      z-index: 100;
      top: 0;
      right: 0;
      width: 100vw;
      height: 100vh;
      background-color: rgba(255,255,255,0.8);
      outline: none;
    }