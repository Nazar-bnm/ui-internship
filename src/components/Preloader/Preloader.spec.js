import React from 'react';
import { shallow } from 'enzyme';

import Preloader from './Preloader';

describe('<Preloader />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Preloader />
    );
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
