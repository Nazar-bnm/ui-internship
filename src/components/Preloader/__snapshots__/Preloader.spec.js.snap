// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> should match snapshot 1`] = `
<div
  className="preloader-wrapper"
>
  <div
    className="preloader"
  >
    <img
      alt="preloder"
      src=""
      style={
        Object {
          "height": undefined,
        }
      }
    />
  </div>
</div>
`;
